extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java

```diff
@@ -34,7 +34,7 @@
     @RequestScoped
     @DefaultBean
     public Uni<Mutiny.Session> mutinySession() {
-        return mutinySessionFactory.openSession();
+        return mutinySessionFactory.openSession().cache();
```

Comment:
I think this is a justifiable limitation of the `Uni<Mutiny.Session>` functionality, because `CompletionStage<Stage.Session>` behaves in the same way, so we make it cached and use the same session in the entire request.

Reply:
It is not a limitation, it is a bug fix :smile: 

The purpose of a request scoped session producer is to have a single session for the request. But if you don't add the `cache` operator, another session will be created anytime someone subscribes to the `Uni` that is injected in the users `beans`.

Reply:
so even if the uni gets subscribed to twice in the same request, that would trigger two different sessions by default? That seems rather confusing, but perhaps necessary since a subscription could happen after the original request has completed?

Reply:
Well, with `.cache` you would in fact get the same session, which is what makes the most sense in this context.